backend-lib/src/main/rust/lib.rs

```diff
@@ -1147,25 +1084,64 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_ge
     unwrap_exc_or(&env, res, -1)
 }
 
-/// Returns a `JniScanProgress` object, provided that numerator is nonnegative, denominator
-/// is positive, and the represented ratio is in the range 0.0 to 1.0 inclusive.
+const JNI_ACCOUNT_BALANCE: &str = "cash/z/ecc/android/sdk/internal/model/JniAccountBalance";
+
+fn encode_account_balance<'a>(
+    env: &JNIEnv<'a>,
+    account: &AccountId,
+    balance: &AccountBalance,
+) -> jni::errors::Result<JObject<'a>> {
+    let sapling_total_balance = Amount::from(balance.sapling_balance.total());
+    let sapling_verified_balance = Amount::from(balance.sapling_balance.spendable_value);
+
+    env.new_object(
+        JNI_ACCOUNT_BALANCE,
+        "(IJJ)V",
+        &[
+            JValue::Int(u32::from(*account) as i32),
```

Comment:
```suggestion
            JValue::Int(u32::from(*account).try_into().unwrap()),
```

Reply:
That is a problem with the `AccountId` type, that should be separately fixed. The rest of the account logic enforces the correct `u31` range, in particular [`WalletDb::create_account`](https://github.com/zcash/librustzcash/blob/e6fa5673328523553e6aa7e083c3fbeaa8dca6fe/zcash_client_sqlite/src/lib.rs#L378-L384). And [casting `u32` to `i32` is a no-op](https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast), so for the `u31` range that account IDs are in, this is exactly what we want.

Reply:
As I said, if account ids are required to be 31 bits, then this is preferable:
```suggestion
            JValue::Int(u32::from(*account).try_into().unwrap()),
```

Reply:
From a semantic perspective these are equivalent. From a codegen perspective, the suggestion results in a bunch of extra code branches that will never be taken, but can't be compiled out. And from a practical perspective, this is already done elsewhere in the FFI, so this is style consistency.